vnext/Desktop.IntegrationTests/HttpResourceIntegrationTests.cpp

```diff
@@ -354,6 +354,104 @@ TEST_CLASS (HttpResourceIntegrationTest) {
     Assert::AreEqual(static_cast<int64_t>(200), getResponse.StatusCode);
     Assert::AreEqual({"Redirect Content"}, content);
   }
+
+  TEST_METHOD(TimeoutSucceeds) {
+    auto port = s_port;
+    string url = "http://localhost:" + std::to_string(port);
+
+    promise<void> getPromise;
+    string error;
+    int statusCode = 0;
+
+    auto server = std::make_shared<HttpServer>(s_port);
+    server->Callbacks().OnGet = [&getPromise](const DynamicRequest &request) -> ResponseWrapper {
+      DynamicResponse response;
+      response.result(http::status::ok);
+
+      // Hold response to test client timeout
+      promise<void> timer;
+      timer.get_future().wait_for(std::chrono::milliseconds(2000));
+
+      return {std::move(response)};
+    };
+    server->Start();
+
+    auto resource = IHttpResource::Make();
+    resource->SetOnResponse([&getPromise, &statusCode](int64_t, IHttpResource::Response response) {
+      statusCode = static_cast<int>(response.StatusCode);
+      getPromise.set_value();
+    });
+    resource->SetOnError([&getPromise, &error, &server](int64_t, string &&errorMessage) {
```

Comment:
>server [](http://example.com/codeflow?start=48&length=6)

Is the lambda doing anything with this capture?

Reply:
Nope. Stale capture.